### ISSUE TITLE
Performance penalty of localConsumer() when in -cluster mode

### ISSUE THREAD

**Issue by javadevmtl** [eclipse/vert.x#1764]
Using vertx 3.3.3

There seems to be a performance penalty when using localConsumer() in clustered mode.
There is cases where you want to mix and match how various verticles are consumed. Some you may only want them as local and others clustered. So a localConsumer() should not incur any performance loss. 

See conversation with Julien and Thomas here: https://groups.google.com/forum/#!topic/vertx/HlWmc4ZuLtA

The reproducer code...
http server: http://pastebin.com/hA4T8Nyy
verticle: http://pastebin.com/wwzRiAut

Basically either run the code as:
1- Non clustered: java -jar ....
2- Clsutered: java  -Dvertx.hazelcast.config=cluster.xml -jar ... -cluster -cluster-host xxx.xxx.xxx

Check the performance between the two commands.

**Comment by tsegismont** [comment]
We need to make sure the behavior I described on the forum is the culprit.

**Comment by vietj** [comment]
@javadevmtl 

1/ can you provide a full reproducer ? i.e a project that one can run
2/ can you check the behavior on older vertx versions like 3.2.x or lower

**Comment by javadevmtl** [comment]
Sure i'll do it asap...

**Comment by javadevmtl** [comment]
Here is the reproducer: https://github.com/javadevmtl/vertx-local-consumer-test

**Comment by javadevmtl** [comment]
@vietj 

I can reproduce your results (you claimed you had slight drop in performance) running on local host only. I.e: the load tester and the vertx app running on my dev box (i5 quad core, 1.6ghz).

It's probably  capping the vertx app by sharing the cpu resources with vertx and the load testing app. I know, you know :)

My dev box (JMeter local, Vertx Local)
----------
Non clustered: 3500 req/sec.
Clustered mode on: 3000 req/sec.

My dev box (JMeter running on separate machine, quad core also)
----------
Non clustered: 6600 req/sec.
Clustered mode on: 6100 req/sec.

Non local (Both JMeter and Vertx running on separate physical machines, both quad cores)
----------
Non clustered: 34000 req/sec.
Clustered mode on: 29000 req/sec

3.2x results...
----------
Non clustered: 34000 req/sec.
Clustered mode on: 29000 req/sec

Each test ran for a few minutes to get past the jit.
